File: ros_time_example/src/time_stamp_example.cpp
#include <rclcpp/rclcpp.hpp>
// #include <geometry_msgs/msg/twist.hpp>
#include <geometry_msgs/msg/twist_stamped.hpp>

namespace ros_time_example {

  class TimeStampExample : public rclcpp::Node {
    public:
      TimeStampExample()
      : rclcpp::Node("time_stamp_example")
      {
        sub_twist_ = create_subscription<geometry_msgs::msg::Twist>("twist", 1, std::bind(&TimeStampExample::recv_twist, this, std::placeholders::_1));
        pub_twist_stamped_ = create_publisher<geometry_msgs::msg::TwistStamped>("twist_stamped", 1);
      }

    private:
      rclcpp::Subscription<geometry_msgs::msg::Twist>::SharedPtr sub_twist_;
      rclcpp::Publisher<geometry_msgs::msg::TwistStamped>::SharedPtr pub_twist_stamped_;

      void recv_twist(const geometry_msgs::msg::Twist::ConstSharedPtr msg) {
        geometry_msgs::msg::TwistStamped twist_stamped_msg;

        twist_stamped_msg.header.stamp = get_clock()->now();
        twist_stamped_msg.twist = *msg;

        pub_twist_stamped_->publish(twist_stamped_msg);
      }

  };

}

int main(int argc, char** argv) {
  rclcpp::init(argc, argv);
  auto node_instance = std::make_shared<ros_time_example::TimeStampExample>();
  rclcpp::spin(node_instance);
  rclcpp::shutdown();
  return 0;
}

File: ros_time_example/scripts/time_stamp_example.py
#! /usr/bin/env python3
import rclpy
from rclpy.node import Node
from geometry_msgs.msg import Twist, TwistStamped


class TimeStampExample(Node):
    def __init__(self):
        super().__init__('time_stamp_example')
        self.sub_twist = self.create_subscription(topic='twist', msg_type=Twist, callback=self.recv_twist, qos_profile=1)
        self.pub_twist_stamped = self.create_publisher(topic='twist_stamped', msg_type=TwistStamped, qos_profile=1)

    def recv_twist(self, msg):
        twist_stamped_msg = TwistStamped()
        twist_stamped_msg.header.stamp = self.get_clock().now().to_msg()
        twist_stamped_msg.twist = msg

        self.pub_twist_stamped.publish(twist_stamped_msg)


if __name__ == '__main__':
    rclpy.init()
    node_instance = TimeStampExample()
    rclpy.spin(node_instance)
